### PR TITLE
Add support for enabling/disabling trigger/condition/action

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

Frontend support for https://github.com/home-assistant/core/pull/70082

CleanShot 2022-04-28 at 13 51 57

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/data/script.ts [hunk 1]
```diff
@@ -104,69 +110,61 @@ export interface ServiceAction {
   data?: Record<string, unknown>;
 }
 
-export interface DeviceAction {
+export interface DeviceAction extends BaseAction {
   alias?: string;
```
Comment (frenck) [inline src/data/script.ts:114]: Already in base?

File: src/panels/config/automation/action/ha-automation-action-row.ts [hunk 2]
```diff
@@ -160,7 +160,14 @@ export default class HaAutomationActionRow extends LitElement {
 
     return html`
       <ha-card>
-        <div class="card-content">
+        ${this.action.enabled === false
+          ? html`<div class="disabled-bar">Disabled</div>`
```
Comment (frenck) [inline src/panels/config/automation/action/ha-automation-action-row.ts:164]: Translatable?

File: src/panels/config/automation/condition/ha-automation-condition-row.ts [hunk 3]
```diff
@@ -59,13 +60,22 @@ export default class HaAutomationConditionRow extends LitElement {
...
+
   protected render() {
     if (!this.condition) {
       return html``;
     }
     return html`
       <ha-card>
-        <div class="card-content">
+        ${this.condition.enabled === false
+          ? html`<div class="disabled-bar">Disabled</div>`
```
Comment (frenck) [inline src/panels/config/automation/condition/ha-automation-condition-row.ts:72]: Translatable?

File: src/panels/config/automation/trigger/ha-automation-trigger-row.ts [hunk 4]
```diff
@@ -126,7 +128,14 @@ export default class HaAutomationTriggerRow extends LitElement {
 
     return html`
       <ha-card>
-        <div class="card-content">
+        ${this.trigger.enabled === false
+          ? html`<div class="disabled-bar">Disabled</div>`
```
Comment (frenck) [inline src/panels/config/automation/trigger/ha-automation-trigger-row.ts:132]: Translatable?

**Review by frenck — CHANGES_REQUESTED** [review]

**Comment by frenck** [comment]
Noticed this:

CleanShot 2022-04-28 at 13 58 09@2x

File: src/panels/config/automation/condition/ha-automation-condition-row.ts [hunk 5]
```diff
@@ -238,6 +273,17 @@ export default class HaAutomationConditionRow extends LitElement {
     return [
       haStyle,
       css`
+        .disabled {
+          opacity: 0.5;
+        }
+        .disabled-bar {
+          background: var(--error-color);
```
Comment (frenck) [inline src/panels/config/automation/condition/ha-automation-condition-row.ts:280]: IMHO it's not an error if done intentionally.

```suggestion
          background: var(--warning-color);
```

File: src/panels/config/automation/trigger/ha-automation-trigger-row.ts [hunk 6]
```diff
@@ -439,6 +472,17 @@ export default class HaAutomationTriggerRow extends LitElement {
     return [
       haStyle,
       css`
+        .disabled {
+          opacity: 0.5;
+        }
+        .disabled-bar {
+          background: var(--error-color);
```
Comment (frenck) [inline src/panels/config/automation/trigger/ha-automation-trigger-row.ts:479]:
```suggestion
          background: var(--warning-color);
```

File: src/panels/config/automation/action/ha-automation-action-row.ts [hunk 7]
```diff
@@ -408,6 +440,17 @@ export default class HaAutomationActionRow extends LitElement {
     return [
       haStyle,
       css`
+        .disabled {
+          opacity: 0.5;
+        }
+        .disabled-bar {
+          background: var(--error-color);
```
Comment (frenck) [inline src/panels/config/automation/action/ha-automation-action-row.ts:447]:
```suggestion
          background: var(--warning-color);
```

**Review by zsarnett — DISMISSED** [review]

File: src/panels/config/automation/condition/ha-automation-condition-row.ts (hunk 5 above)
Comment (frenck) [reply src/panels/config/automation/condition/ha-automation-condition-row.ts:285]: I must agree I've seen better... 

image

This is how we show disabled in the integrations dashboard, maybe take that?

**Pull request merged**